### DISCUSSION TITLE
Bot hosting bandwidth

### DISCUSSION THREAD

**Discussion by Alefunshow** [jagrosh/MusicBot#1321] [Questions]
I wanted to know how much bandwidth it consumes approximately when It is used from around 10 people.
Thank you very much!

**Comment by ghost** [comment]
The bot transmits the data to the discord server and that server then forwards the audio to each client. So additional listeners in the channel won‘t increase the bandwidth. It wouldn‘t make a difference if there would be 100 people listening. 

The bandwidth itself should be what you set your discord channel too. If you didn‘t change it it should 64 kbit/s. So that‘s around 30 megabytes for an hour of uninterrupted playback. However depending on the source of the audio you are playing the real traffic could be actually a lot higher. If you play your audio from youtube, the bot will need to download all the youtube source material.

**Reply by jagrosh** [reply]
My knowledge of the inner workings of lavaplayer and jda's old audio code might be outdated, but as far as I know, the bot will always pass along the audio data at the source's bitrate, so playing a higher-quality audio stream will cost more bandwidth (both incoming to the bot and outgoing to discord) regardless of what the channel is set to.

First part is correct though; the bot isn't streaming to each person, it's streaming to Discord, so having more people listening isn't increasing the bandwidth in that regard.
